### PR TITLE
Handle get token request for che.openshift.io

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Handle `getToken` request in `che.openshift.io` according to oAut API in `che.openshift.io`.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16890

### CODE REVIEW

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts [hunk 1]
```diff
@@ -24,13 +27,25 @@ export class CheOpenshiftMainImpl implements CheOpenshiftMain {
...
+            }
+        });
     }
     async $getToken(): Promise<string> {
         let oAuthProvider = 'openshift';
         const unauthorisedMessage = 'Request failed with status code 401';
         // Multi-user mode doesn't support list of registered provers request,
         // so we need to check which version of openshift is registered.
         if (this.isMultiUser) {
+            if (this.isOpenshiftIO) {
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts:42]: `isHostedChe` sounds more general

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts [hunk 2]
```diff
@@ -24,13 +27,25 @@ export class CheOpenshiftMainImpl implements CheOpenshiftMain {
                 this.isMultiUser = true;
             }
         });
+        container.get<EnvVariablesServer>(EnvVariablesServer).getValue('CHE_API').then(variable => {
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts:30]: how about getting `EnvVariablesServer` once and store it in a variable?
I believe inversifyjs is fast enough ) but less code duplication, at least

**Comment by ericwill** [comment]
> +1 could we make it available in 7.15.0 ?

Igor is on PTO at the moment -- if everyone is okay with this change (apart from small improvements) then sure we can merge it.

**Comment by azatsarynnyy** [comment]
let's re-run HappyPath on it

**Comment by azatsarynnyy** [comment]
crw-ci-test

**Comment by azatsarynnyy** [comment]
crw-ci-test

**Comment by ericwill** [comment]
crw-ci-test

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts (hunk 1 above)
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts]: fixed

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts (hunk 2 above)
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/che-openshift-main.ts]: fixed

**Pull request merged**